blog/2024-08-06-wasmcloud-1.1-brings-secrets-support-and-more-enhancements-for-enterprises/index.mdx

```diff
@@ -0,0 +1,53 @@
+image: './images/wasmcloud-1-1.png'
+authors: [brooksmtownsend]
+date: "2024-08-06"
+description: 'Learn about the new features in wasmCloud 1.1, including secrets support.'
+slug: wasmcloud-1.1-brings-secrets-support-and-more-enhancements-for-enterprises
+---
+
+![header](./images/wasmcloud-1-1.png)
+
+We're proud to announce the release of **wasmCloud 1.1**, including the feature most commonly requested by users: **secrets support**. In addition to secrets, the new release brings along a bevy of enhancements for enterprises, exciting new capabilities like Postgres, and first-class support for Go-based custom capability providers. Let's take a look at some of the highlights.
+
+{/* truncate */}
+
+## Secrets in wasmCloud
+
+Secrets are a must-have feature for enterprises, so it's no surprise that secret support has been our most-requested feature since the launch of 1.0. Now we're thrilled to provide full and flexible support for secrets with a secrets **backend system** that enables you to use the secret store of your choice while maintaining robust encryption across the lifecycle of a secret. 
+
+Here you can see a high-level diagram of how a secret request travels from a component to two different secret stores: HashiCorp Vault and a NATS key-value store. (Not coincidentally, secrets backends are already implemented for these two stores.)  
+
+![diagram](./images/secrets-diagram.png)
+
+wasmCloud also integrates fully with Kubernetes secrets, so you can make use of any existing Kubernetes secrets infrastructure right out of the box, if you're [running wasmCloud on Kubernetes](/blog/how-to-run-webassembly-components-on-kubernetes-with-wasmcloud). 
+
+For a full overview of how secrets work in wasmCloud, [check out the documentation](/docs/concepts/secrets), which includes a [hands-on example of using secrets with a NATS key-value store](/docs/concepts/secrets#example-using-a-nats-kv-backend). You can also [learn how to implement your own secrets backend](/docs/deployment/security/secrets).
+
+## Postgres and Couchbase
+
+wasmCloud maintainers and the wider community have released several major capabilities in the process of landing wasmCloud 1.1. There are two capabilities we'd particularly like to highlight&mdash;each of which can leverage our new secrets functionality for secure authentication:
+
+* **Postgres**: The first-party PostgreSQL [interface](https://github.com/wasmCloud/wasmCloud/tree/main/wit/postgres) and [capability provider](https://github.com/wasmCloud/wasmCloud/pkgs/container/sqldb-postgres) combine to bring Postgres functionality to wasmCloud applications as a simple, reusable building block. 
+* **Couchbase**: The third-party Couchbase [interface](https://github.com/couchbaselabs/wasmcloud-provider-couchbase/tree/main/wit/couchbase) and [provider](https://github.com/couchbaselabs/wasmcloud-provider-couchbase) bring Couchbase KV connectivity to Wasm applications via `wasi-keyvalue`.
+
```

Comment:
Should we have a quick little title here for Go providers?

```suggestion
## Go Capability Providers

During the development of wasmCloud 1.1 the [provider-sdk-go](https://github.com/wasmCloud/provider-sdk-go) library got many enhancements and improvements, including the addition of secrets support. In the cloud-native ecosystem there are a wealth of open source libraries written in Go that you can now use to build custom capability providers for wasmCloud. Just like in Rust, bindings for Go providers are generated via [wRPC](https://github.com/bytecodealliance/wrpc). You can see a examples of Go capability providers either in the Couchbase provider above, or the [custom template](https://github.com/wasmCloud/wasmCloud/tree/main/examples/golang/providers/custom-template) provider for Go.

```bash
wash new provider --template-name custom-template-go go-provider
```

```